plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -137,16 +138,55 @@ export class ActionMatcher {
         this.statsd = statsd
     }
 
-    /** Get all actions matched to the event. */
     public async match(event: PostIngestionEvent, elements?: Element[]): Promise<Action[]> {
+        const legacyResponse = await this.matchLegacy(event, elements)
+        const hogVMResponse = this.matchBytecode(event, elements)
+
+        if (
+            legacyResponse.length !== hogVMResponse.length ||
+            !legacyResponse.every((action, index) => action.id == hogVMResponse[index]?.id)
+        ) {
+            this.statsd?.increment('action_matches_hogvm_legacy_mismatch')
+        } else {
+            this.statsd?.increment('action_matches_hogvm_legacy_match')
+        }
+        console.log({ hogVM: hogVMResponse.map((a) => a?.id), legacy: legacyResponse.map((a) => a?.id) })
+
+        return legacyResponse
+    }
+
+    /** Get all actions matched to the event. */
+    public matchBytecode(event: PostIngestionEvent, elements?: Element[]): (Action | undefined)[] {
+        const matchingStart = new Date()
+        const teamActions: Action[] = Object.values(this.actionManager.getTeamActions(event.teamId))
+        if (!elements) {
+            const rawElements: Record<string, any>[] | undefined = event.properties?.['$elements']
+            elements = rawElements ? extractElements(rawElements) : []
+        }
+        const teamActionsMatching: (boolean | undefined)[] = teamActions.map((action) =>
+            this.checkActionBytecode(event, elements, action)
+        )
+
+        const matches: Action[] = []
+        for (let i = 0; i < teamActionsMatching.length; i++) {
+            if (teamActionsMatching[i]) {
+                matches.push(teamActions[i])
+            }
+        }
+        this.statsd?.timing('action_matching_for_event_hogvm', matchingStart)
+        this.statsd?.increment('action_matches_found_hogvm', matches.length)
```

Comment:
Isn't timing functionally a superset of increment? As in, I believe we can count the number of `action_matching_for_event_hogvm`.

Also, it's interesting to have the timing data for bytecode, but it would be _most_ interesting to compare bytecode and legacy matching, which requires analogous instrumentation in `checkActionLegacy`